compose/ui/ui-backhandler/src/jbMain/kotlin/androidx/compose/ui/backhandler/BackHandler.jb.kt

```diff
@@ -41,15 +43,24 @@ actual fun PredictiveBackHandler(
     onBack: suspend (progress: Flow<BackEventCompat>) -> Unit
 )  {
     val backGestureDispatcher = LocalBackGestureDispatcher.current ?: return
-    val lifecycleOwner = LocalLifecycleOwner.current
-    val scope = rememberCoroutineScope()
+
+    // ensure we don't re-register callbacks when onBack changes
+    val currentOnBack by rememberUpdatedState(onBack)
+    val onBackScope = rememberCoroutineScope()
 
     val listener = remember {
-        BackGestureListenerImpl(scope, onBack) { backGestureDispatcher.activeListenerChanged() }
+        BackGestureListenerImpl(onBackScope, currentOnBack) { backGestureDispatcher.activeListenerChanged() }
+    }
+
+    // we want to use the same callback, but ensure we adjust the variable on recomposition
+    remember(currentOnBack, onBackScope) {
+        listener.currentOnBack = currentOnBack
+        listener.onBackScope = onBackScope
     }
```

Comment:
I'm not sure this is wrong, but it's not the usual way of doing things.

1. I don't think `rememberCoroutineScope()` ever changes, so there's no need to update it.
2. For `onBack`, either create a new `BackGestureListenerImpl`, or pass it a function that doesn't change:
```
BackGestureListenerImpl(
    scope = scope,
    onBack = { currentOnBack(it) }
)
```

By the way, you need to do the same for `backGestureDispatcher`, as it could also change.

Reply:
I don't catch the `backGestureDispatcher`. it is callback

Reply:
1) you are right about the coroutine scope - I fixed it.
2) the code reuses the same instance of BackGestureListenerImpl and updates callback inside. this is a copy of the android code

Reply:
> I don't catch the backGestureDispatcher. it is callback

You're capturing it in
```
BackGestureListenerImpl(onBackScope, currentOnBack) { backGestureDispatcher.activeListenerChanged() }
```
but you're not updating it if it changes.

Reply:
a-ah! I see

Reply:
done